fuzz_test.go

```diff
@@ -0,0 +1,103 @@
+package smt
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func FuzzSMT(f *testing.F) {
+	seeds := [][]byte{
+		[]byte(""),
+		[]byte("foo"),
+		{1, 2, 3, 4},
+		[]byte("\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"),
+		nil,
+	}
+	for _, s := range seeds {
+		f.Add(s)
+	}
+	f.Fuzz(func(t *testing.T, input []byte) {
+		t.Log(input)
+		smn := NewSimpleMap()
+		tree := NewSparseMerkleTree(smn, sha256.New())
+
+		r := bytes.NewReader(input)
+		var keys [][]byte
+		key := func() []byte {
```

Comment:
Can you add a few comments on what this function does. I realize you just want a random key with bytes, but something like:

With a 50% chance, generate a new key and return it.
Otherwise, return an existing random key if any exist.